cli/cmd/deploy/deploy.go

```diff
@@ -361,6 +380,16 @@ func verifyAccess(ctx context.Context, c *client.Client, githubURL string) (*adm
 				return pollRes, nil
 			}
 
+			// open user authorisation url once per invocation
+			if pollRes.UserAuthorisationUrl != "" && !userAuthRequested {
+				userAuthRequested = true
+				fmt.Printf("Open this URL in your browser to grant Rill access to your account:\n\n")
+				fmt.Printf("\t%s\n\n", pollRes.UserAuthorisationUrl)
+				time.Sleep(1 * time.Second)
+				// Open browser if possible
+				_ = browser.Open(ghRes.GrantAccessUrl)
+			}
```

Comment:
I don't understand why this case is needed. It also seems weird to open the browser suddenly during the polling?

Also, it checks `pollRes.UserAuthorisationUrl`, but opens `ghRes.GrantAccessUrl`

Reply:
Without maintaining a state on backend we have no way of telling the user that they requested access and auto-terminate the cli. The cli can now terminate in following states in case of `request`:
1. CLI times out
2. User cancels the CLI on seeing the github page that requested access.
3. Admin approves the request within timeout period, we navigate the user on authorisation page and user successfully deploys the project.

I agree case 3 is not ideal and we can remove this case. A user will be expected to re-run `rill deploy` in all cases. 
Opening `ghRes.GrantAccessUrl` is a bug.